__init__.py

```diff
@@ -362,16 +362,33 @@ def selectLanguage( self, path ):
 			tmppath = [ urlparse.unquote( x ) for x in tmppath.lower().strip("/").split("/") ]
 			if len( tmppath )>0 and tmppath[0] in conf["viur.availableLanguages"]+list( conf["viur.languageAliasMap"].keys() ):
 				self.language = tmppath[0]
-				return( path[ len( tmppath[0])+1: ] ) #Return the path stripped by its language segment
+				return path[ len( tmppath[0])+1: ]  # Return the path stripped by its language segment
 			else: # This URL doesnt contain an language prefix, try to read it from session
-				if session.current.getLanguage():
-					self.language = session.current.getLanguage()
-				elif "X-Appengine-Country" in self.request.headers.keys():
+				sessionLang = session.current.getLanguage()
+				if sessionLang:
+					self.language = sessionLang
+					return path
+				if "Accept-Language" in self.request.headers:
+					acceptLangHeader = self.request.headers["Accept-Language"]
+					if acceptLangHeader:
+						acceptLangHeader = acceptLangHeader.split(",")
+						# we only accept up to seven language entries here.
+						for possibleLang in acceptLangHeader[:7]:
+							lng = possibleLang.split("-")[0]
+							if lng in conf["viur.availableLanguages"] or lng in conf["viur.languageAliasMap"]:
+								self.language = lng
+								return path
+							elif ";" in lng:
+								lng, weight = lng.split(";")
```

Comment:
Just one more small thing. This Line can also fail. And you don't use weight at all. So also use lng.split(";")[0] - which doesn't break if there are more than two colons in that string